src/getPublicKey.c

```diff
@@ -106,7 +121,7 @@ static unsigned int io_seproxyhal_touch_pk_ok(void) {
     // variable is named 'tx'.
     uint8_t tx = 0;
 
-    deriveSiaKeypair(ctx->keyIndex, NULL, &publicKey);
+    deriveSiaPublicKey(ctx->keyIndex, publicKey.W);
```

Comment:
You can directly allocate a buffer of 65 bytes instead of a `cx_ecfp_public_key_t` here.
Then you would have to change the prototype of `extractPubkeyBytes` and `pubkeyToSiaAddress`, but that for a good thing as you shouldn't have to use the internals of `cx_ecfp_public_key_t` structure.